src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const WHITE_SPACE_REGEX = /\s+/g;
+  const cssObject = {};
+
+  const trimmedLines = sourceString.split(';')
+    .map((line) =>
+      line.replace(WHITE_SPACE_REGEX, ' ')
+        .trim())
+    .filter((line) => line.trim() !== '');
+
+  trimmedLines.forEach((pair) => {
+    const [property, value] = pair.split(':').map((item) => item.trim());
```

Comment:
```suggestion
    const [property, value] = pair
      .split(':')
      .map((item) => item.trim());
```
![image](https://github.com/mate-academy/js_style-to-object/assets/96409563/77acd0cc-9a0e-4645-9c96-25729d843a47)